src/udev/udevadm-test-builtin.c

```diff
@@ -29,15 +31,23 @@ static int help(void) {
 
 static int parse_argv(int argc, char *argv[]) {
         static const struct option options[] = {
-                { "version", no_argument, NULL, 'V' },
-                { "help",    no_argument, NULL, 'h' },
+                { "action",  required_argument, NULL, 'a' },
+                { "version", no_argument,       NULL, 'V' },
+                { "help",    no_argument,       NULL, 'h' },
                 {}
         };
 
-        int c;
+        int r, c;
 
-        while ((c = getopt_long(argc, argv, "Vh", options, NULL)) >= 0)
+        while ((c = getopt_long(argc, argv, "aVh", options, NULL)) >= 0)
```

Comment:
Shouldn't this be "a:Vh" since the action has a required argument? I got an assertion in parse_device_action while trying this out because `str` was `NULL`. Changing the option string to "a:Vh" fixes the issue

Reply:
Ouch.